client/dashboard/app/queries.ts

```diff
@@ -98,3 +101,47 @@ export function profileMutation() {
 		},
 	};
 }
+
+export function siteSettingsQuery( siteId: string ) {
+	return {
+		queryKey: [ 'site-settings', siteId ],
+		queryFn: () => {
+			return fetchSiteSettings( siteId );
+		},
+		refetchOnWindowFocus: false,
+		retry: false,
+		enabled: !! siteId,
+	};
+}
+
+export function basicMetricsQuery(
+	url: string,
+	isLoadingSiteSettings: boolean,
+	cachedHash: string
+) {
+	return {
+		queryKey: [ 'url', 'basic-metrics', url ],
+		queryFn: () => {
+			return fetchBasicMetrics( url );
+		},
+		refetchOnWindowFocus: false,
+		enabled: !! url && ! isLoadingSiteSettings && ! cachedHash,
+	};
+}
+
+export function performanceInsightsQuery( url: string, token: string ) {
+	return {
+		queryKey: [ 'url', 'performance', url, token ],
+		queryFn: () => {
+			return fetchPerformanceInsights( url, token );
+		},
+		enabled: !! url && !! token,
```

Comment:
Same for these

Reply:
Same [as comment](https://github.com/Automattic/wp-calypso/pull/102956/files#r2067554091), pre-dates PR. Its logical to enable/disable based on the availability of the URL and token. It's a common React Query pattern to avoid necessary conditional checks upstream.

Reply:
I know it pre-dates the PR. What I mean is that I don't think it belong here, I think we should add these options wherever the query is used?